### ISSUE TITLE
Should <require> allow wildcards?

### ISSUE THREAD

**Issue by jsobell** [aurelia/templating#548]
With inclusion of `PLATFORM.moduleName` we have a cleaner means of parsing bundle dependencies from things like the webpack loader, but since webpack supports dynamic contexts, how do we include that information at the view level?
We can add plugins to glob in the folder contents, but that puts the view dependencies in the loader, rather than associating them with the component.
Would there be any issue with something like
```html
<require from="elements/status-*"></require>
```
and to extend the webpack (or any other framework) loader to detect wild card use properly?
I suppose the alternative is to do it 'properly' and allow
```html
<compose view="elements/status-${status}"></compose>
```
and have the loader ensure `elements/status-*` is used to wildcard in the inclusions. This is in line with nature of webpack dynamic contexts, but would it cause issues with other packaging options?

**Comment by niieani** [comment]
Seems like a good idea. Thoughts @jods4?

**Comment by EisenbergEffect** [comment]
Won't work with other module loaders including the whatwg standard.

**Comment by jods4** [comment]
I feel like there are two different discussion in the issue description.

Regarding the title one: _Should <require> allow wildcards?_
I will defer this to Rob as the first step is specify the behavior of `aurelia-templating` when `<require from="folder/*">` is encountered.
Adding support in Webpack would come second after implementation in Aurelia.

My personal opinion on that:
- Loaders are not able to list contents of a folder, so I don't see how that could be defined/work.
- What extensions would be grabbed? Pretty much _any_ extension could be valid for someone.
- ES standard doesn't need `import "folder/*"` or `import("folder/*")`. Why would we? + being explicit brings more safety.

:-1:

**Comment by niieani** [comment]
The arguments against are pretty good, I'm persuaded. We should probably mention in the blog post that we deprecated dynamic imports in Webpack though (explained here https://github.com/aurelia/skeleton-navigation/issues/810), unless that's something we want to consider implementing.

**Comment by jods4** [comment]
I did not see that issue. I'll leave a comment there.
It's not deprecated as there are good use cases to bind `<compose>`. 
The way you handle dependencies has changed, though. Auto-include everything doesn't work anymore.

**Comment by jsobell** [comment]
I have no issue with webpack's dynamic context in module 'require' functions being deprecated, but then I've never wanted to use it because it looks like a messy shortcut to referring to dependencies properly.
Other people might feel differently, so we'll need to get some feedback.

As @jods4 says, it's really about composition, and how a loader is able to say "this template has a dependency on *this* template or *these* templates".
I'm happy to insist each 'child' dependency template be on a separate line, but if so we need to lock it in and document that as a tooling change.

I'll close https://github.com/aurelia/skeleton-navigation/issues/810#issuecomment-297869833 and point people to this discussion, as I believe the template level needs to be the lowest common denominator between loaders.

**Comment by istrau3** [comment]
So what are the solutions for someone who wants to use dynamic composition such as
```
<compose view="elements/status-${status}"></compose>
```
import all those modules explicitly using require statements?

**Comment by jods4** [comment]
@istrau3 I'll add a page to the wiki about it.

In the meantime all options are described here:
https://github.com/aurelia/skeleton-navigation/issues/810#issuecomment-297760384

TL;DR: it's much better to not use interpolation but put the full module id in your code, wrapped in `moduleName`. It has several benefits in large-scale projects, see the discussion that followed the comment linked above.

**Comment by EisenbergEffect** [comment]
Closing this since it's not something we can do. I wish we could.
